tests/nn/data_parallel/test_fsdp.py

```diff
@@ -113,6 +113,10 @@ def _test_identical_outputs(
         else:
             assert next(model.parameters()).device == torch.device("cpu")
         shard_loss = cls._train_for_several_steps(model, num_steps, autocast, lr=lr, norm_type=norm_type)
+        if config.get("cpu_offload", False):
```

Comment:
I encountered the same error while upstreaming FSDP to PyTorch, I added 'ref_loss = ref_loss.cpu()' after line 105, it works

Reply:
Good to know! I just wanted to make sure we only move the device when we are expecting in the test so that it is a bit more strict.